packages/opentelemetry-resources/src/detectors/platform/node/ProcessDetector.ts

```diff
@@ -14,18 +14,54 @@
  * limitations under the License.
  */
 
-import { Detector } from '../../../types';
+import { Attributes, diag } from '@opentelemetry/api';
+import {
+  SEMRESATTRS_PROCESS_COMMAND,
+  SEMRESATTRS_PROCESS_COMMAND_ARGS,
+  SEMRESATTRS_PROCESS_EXECUTABLE_NAME,
+  SEMRESATTRS_PROCESS_EXECUTABLE_PATH,
+  SEMRESATTRS_PROCESS_OWNER,
+  SEMRESATTRS_PROCESS_PID,
+  SEMRESATTRS_PROCESS_RUNTIME_DESCRIPTION,
+  SEMRESATTRS_PROCESS_RUNTIME_NAME,
+  SEMRESATTRS_PROCESS_RUNTIME_VERSION,
+} from '@opentelemetry/semantic-conventions';
+import * as os from 'os';
 import { ResourceDetectionConfig } from '../../../config';
-import { IResource } from '../../../IResource';
-import { processDetectorSync } from './ProcessDetectorSync';
+import { ResourceDetector } from '../../../types';
 
 /**
  * ProcessDetector will be used to detect the resources related current process running
  * and being instrumented from the NodeJS Process module.
  */
-class ProcessDetector implements Detector {
-  detect(config?: ResourceDetectionConfig): Promise<IResource> {
-    return Promise.resolve(processDetectorSync.detect(config));
+class ProcessDetector implements ResourceDetector {
+  detect(_config?: ResourceDetectionConfig) {
```

Comment:
`: DetectedResource` ...if we want to be consistent about public interface return types, I'd leave a suggestion but this requires adding the import too